### PR TITLE
🐛Introduce timeout in project lock (a la aioredlock)

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
Previously we were using aioredlock to lock projects. When we migrated to redis-py, we lost the automatic timeout extension of distributed locks. This probably led to the following issues:
https://github.com/ITISFoundation/osparc-issues/issues/724
https://github.com/ITISFoundation/osparc-simcore/issues/3145

in case of a crash in the webserver, these locks would never timeout.

This PR aims to fix this by mimicking the behavior of aioredlock (default timeout of 10 seconds, with auto-extension every 6 seconds). In case the webserver would restart/crash without having the time to release the lock, this would ensure the lock is released at some point.
This PR just fixes the ProjectLock, all the other locks are not concerned with this change.

just looking at master's redis-commander shows how this is a problem, most of these locks are lost:
image

### CODE REVIEW

File: services/web/server/tests/unit/with_dbs/03/test_redis.py [hunk 1]
```diff
@@ -44,13 +76,44 @@ async def test_redlocks_features(redis_client: aioredis.Redis):
...
+        _auto_extend_lock,
+        interval=0.6 * lock_timeout,
+        task_name=f"{ttl_lock.name}_auto_extend",
+        lock=ttl_lock,
+    ):
+        assert await ttl_lock.locked()
+        assert await ttl_lock.owned()
+        await asyncio.sleep(5 * lock_timeout.total_seconds())
+        assert await ttl_lock.locked()
+        assert await ttl_lock.owned()
```
Comment (GitHK) [inline services/web/server/tests/unit/with_dbs/03/test_redis.py:119]: Question: Would it make sense to also test for the lock being released when exiting the context manager?

**Review by GitHK — APPROVED** [review]
Very nice! Thanks for addressing this!

File: packages/service-library/src/servicelib/background_task.py [hunk 2]
```diff
@@ -64,3 +64,22 @@ async def stop_periodic_task(
         asyncio_task.cancel()
         with log_catch(logger, reraise=False):
             await asyncio.wait((asyncio_task,), timeout=timeout)
+
+
+@contextlib.asynccontextmanager
+async def periodic_task(
```
Comment (pcrespov) [inline packages/service-library/src/servicelib/background_task.py:70]: MINOR: something in the name highlight that this is a context manager? e.g. ``periodic_task_context`` or ``periodic_task_lifespam``?

**Review by pcrespov — APPROVED** [review]
👍

File: packages/service-library/src/servicelib/background_task.py (hunk 2 above)
Comment (sanderegg) [reply packages/service-library/src/servicelib/background_task.py:70]: I am actually pretty happy with this name, and pylance nicely show it is a context manager.

**Pull request merged**